app/Controllers/extensionController.php

```diff
@@ -42,7 +42,17 @@ public function indexAction() {
 	 */
 	protected function getAvailableExtensionList() {
 		$extensionListUrl = 'https://raw.githubusercontent.com/FreshRSS/Extensions/master/extensions.json';
-		$json = @file_get_contents($extensionListUrl);
+
+		$cacheFile = CACHE_PATH . '/extensions.json';
+		if (FreshRSS_Context::$user_conf->retrieve_extension_list === true) {
+			if (!file_exists($cacheFile) || (time() - filemtime($cacheFile) > 86400)) {
+				@file_put_contents($cacheFile, file_get_contents($extensionListUrl));
+			}
+			$json = @file_get_contents($cacheFile);
```

Comment:
warning : **don't use the silent operator @, you have to handle the errors properly**

Reply:
It is addressed 5 lines below. The `@` is correct here to avoid the risk of showing a warning to end-users